Framework/MDAlgorithms/src/CreateMDHistoWorkspace.cpp

```diff
@@ -98,44 +98,27 @@ void CreateMDHistoWorkspace::exec() {
   auto errors = ws->mutableErrorSquaredArray();
   auto nEvents = ws->mutableNumEventsArray();
 
-  std::vector<double> signalValues = getProperty("SignalInput");
-  std::vector<double> errorValues = getProperty("ErrorInput");
-  std::vector<double> numberOfEvents = getProperty("NumberOfEvents");
+  const std::vector<double> &signalValues = getProperty("SignalInput");
+  const std::vector<double> &errorValues = getProperty("ErrorInput");
+  const std::vector<double> &numberOfEvents = getProperty("NumberOfEvents");
 
   size_t binProduct = this->getBinProduct();
-  std::stringstream stream;
-  stream << binProduct;
-  if (binProduct != signalValues.size()) {
-    throw std::invalid_argument("Expected size of the SignalInput is: " + stream.str());
-  }
-  if (binProduct != errorValues.size()) {
-    throw std::invalid_argument("Expected size of the ErrorInput is: " + stream.str());
-  }
-  if (!numberOfEvents.empty() && binProduct != numberOfEvents.size()) {
-    throw std::invalid_argument("Expected size of the NumberOfEvents is: " + stream.str() +
-                                ". Leave empty to auto fill with 1.0");
+  if (signalValues.size() != binProduct || errorValues.size() != binProduct ||
+      (!numberOfEvents.empty() && numberOfEvents.size() != binProduct)) {
+    throw std::invalid_argument("All inputs must match size: " + std::to_string(binProduct));
   }
 
-  // Auto fill number of events.
+  // Fast memory copies and squaring
+  std::memcpy(signals, signalValues.data(), binProduct * sizeof(double));
+  for (size_t i = 0; i < binProduct; ++i)
```

Comment:
I think this would be subjectively better with [std::transform](https://en.cppreference.com/w/cpp/algorithm/transform.html) (assuming `errors` has already allocated the right amount of space)
```cpp
std::transform(errorValues.cbegin(), errorValues.cend(), errors.begin(),
               [](const auto &value){return value*value;});
```
the `Square` struct earlier in the file could be removed